packages/gatsby/cache-dir/head/head-export-handler-for-ssr.js

```diff
@@ -62,14 +62,28 @@ export function headHandlerForSSR({
       if (!VALID_NODE_NAMES.includes(rawTagName)) {
         warnForInvalidTags(rawTagName)
       } else {
-        const element = createElement(
-          rawTagName,
-          {
-            ...attributes,
-            "data-gatsby-head": true,
-          },
-          node.childNodes[0]?.textContent
-        )
+        let element
+        // Special handling for scripts with data block to prevent React from escaping the text
+        if (
```

Comment:
I don't like the special handling of things here, it should all go through the same code path because there will surely be cases where this is needed and it's not a JSON-LD script